sympy/functions/elementary/exponential.py

```diff
@@ -456,6 +456,16 @@ def _eval_rewrite_as_tanh(self, arg):
         from sympy import tanh
         return (1 + tanh(arg/2))/(1 - tanh(arg/2))
 
+    def _eval_rewrite_as_sqrt(self, arg):
+        from sympy.functions.elementary.trigonometric import sin, cos
+        if arg.is_Mul:
+            coeff = arg.coeff(S.Pi*S.ImaginaryUnit)
+            cosine, sine = cos(S.Pi*coeff), sin(S.Pi*coeff)
+            if not isinstance(cosine, cos) and not isinstance (sine, sin):
+                return cosine + S.ImaginaryUnit*sine
+        elif (arg.is_Mul == False):
```

Comment:
It seems that None can be returned to indicate no rewrite. Hence this else part can be omitted.
But there is something else that should be fixed.
```
>>> x = symbols('x')
>>> exp(x*I).rewrite(sqrt)
1
```
There should be more tests to cover different cases. Also this one on the original issue could be included.
```
>>> Sum((exp(pi*I/2)/2)**n, (n, 0, oo)).rewrite(sqrt).doit()
4/5 + 2*I/5
```

Reply:
@jksuom why is `exp(x*I).rewrite(sqrt)` equal to `1` ?

Reply:
It looks like what would result when `coeff == 0`.